lib/src/track/visualizer.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:livekit_client/src/support/disposable.dart';
+import '../events.dart' show AudioVisualizerEvent;
+import '../managers/event.dart' show EventsEmittable;
+import 'local/local.dart' show AudioTrack;
+
+import 'visualizer_native.dart'
+    if (dart.library.js_interop) 'visualizer_web.dart';
+
+class VisualizerOptions {
+  final bool centeredBands;
+  final int barCount;
+  const VisualizerOptions({
+    this.centeredBands = true,
+    this.barCount = 7,
+  });
+}
+
+abstract class Visualizer extends DisposableChangeNotifier
```

Comment:
the events are called `AudioVisualizerEvents`, maybe it makes sense to call the Visualizer also `AudioVisualizer` for clarity? 

JS's naming is explicit with `BarVisualizer` as we might add more/different visualizers down the road.